api4/service_terms.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package api4
+
+import (
+	"github.com/mattermost/mattermost-server/app"
+	"github.com/mattermost/mattermost-server/model"
+	"net/http"
+)
+
+func (api *API) InitServiceTerms() {
+	api.BaseRoutes.ServiceTerms.Handle("", api.ApiSessionRequired(getServiceTerms)).Methods("GET")
+	api.BaseRoutes.ServiceTerms.Handle("", api.ApiSessionRequired(createServiceTerms)).Methods("POST")
+}
+
+func getServiceTerms(c *Context, w http.ResponseWriter, r *http.Request) {
+	serviceTerms, err := c.App.GetLatestServiceTerms()
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	w.Write([]byte(serviceTerms.ToJson()))
+}
+
+func createServiceTerms(c *Context, w http.ResponseWriter, r *http.Request) {
+	props := model.MapFromJson(r.Body)
+	text := props["text"]
+	userId := c.Session.UserId
+
+	if text == "" {
+		c.Err = model.NewAppError("Config.IsValid", "api.create_service_terms.empty_text.app_error", nil, "", http.StatusBadRequest)
+	}
+
+	oldServiceTerms, err := c.App.GetLatestServiceTerms()
```

Comment:
Now that this isn't gated by the config permissions, I think we need an explicit permission check.

Reply:
Done. Returned 403 if feature is disabled,